app/models/user.rb

```diff
@@ -101,4 +102,11 @@ def time_zone_identifier
     member: 0,
     admin: 1
   }
+
+  private
+
+  def update_profile_slug!
```

Comment:
kind of confused as to why we do this after update? Is this to force it to update the slug? Users should be able to choose their own slug and make sure it is uniq across all profiles (we shouldn't be doing any magic really).

Also for testing this because you did ask, you'd want to do something that forces this to be used (i.e: make an update, which would then trigger this, and check the record to see it is updated as expected)

Reply:
not sure if Daniel would want it in this issue, but I love the idea of the "denylist of words" feat. Are there any gems/services that would help with that?

Reply:
For the default, we can make it something automatically populated (i'd even say maybe `first_name.last_name` for the initial).

For characters we should allow, do something like all alphanumeric, period, hyphen and underscore (.-_). That should be good enough for now.

As for denylist, we should prob find a gem to help with that. For now we won't add anything.

Reply:
> For the default, we can make it something automatically populated (i'd even say maybe `first_name.last_name` for the initial).
> 
> For characters we should allow, do something like all alphanumeric, period, hyphen and underscore (.-_). That should be good enough for now.
> 
> As for denylist, we should prob find a gem to help with that. For now we won't add anything.

In terms of allowed characters, I don't think we can allow `.` as that typically denotes a format in a request URL. How do you feel about alphanumeric with just `-_` as well?